### ISSUE TITLE
Tab character indentation option for Google Colab

### ISSUE THREAD

**Issue by baxterai** [googlecolab/colabtools#2431]
**Is your feature request related to a problem? Please describe.**
Quickly moving and clicking the mouse at a position within the indentation (e.g. start or end of indentation) should not result in the cursor being moved to an arbitrary, redundant space within the indentation.

**Describe the solution you'd like**
Tab character indentation does not have this problem - it allows for a certain amount of horizontal error in fast GUI operations.

**Describe alternatives you've considered**
N/A

**Additional context**
Providing such an option is also consistent with the features offered by other text editors (cross-compatibility).

**Comment by blois** [comment]
If I understand this correctly- mouse clicking within the leading indentation should round to tab stops?

Is there an example of another editor which does this, when spaces are used as tabs?

**Comment by baxterai** [comment]
Yes this is the ideal behaviour (and provided by editors that support tab character indentation).

Likewise pressing the arrow keys should not result in the cursor being moved to an intermediate space within the indentation, but between the emulated tab stops (x spaces).

It is likely that all/most text editors have the same issues with emulated tab stops (tested; NEdit, VS, Textpad, Notepad++).

**Comment by bairesearch** [comment]
Colab now automatically detects and supports tab character indentation in cells.

**Comment by bairesearch** [comment]
> Colab now automatically detects and supports tab character indentation in cells.

Colab does not currently autodetect indentation type (tabs/spaces) in opened files, only in the main notebook.
